src/dvc_data/hashfile/hash.py

```diff
@@ -190,5 +190,9 @@ def hash_file(
         assert ".dir" not in hash_info.value
         state.save(path, fs, hash_info)
 
-    meta = Meta(size=info["size"], isexec=is_exec(info.get("mode", 0)))
+    meta = Meta(
+        size=info["size"],
+        isexec=is_exec(info.get("mode", 0)),
+        version_id=info.get("version_id", info.get("VersionId")),
```

Comment:
Looking at adlfs again, azure uses `versionid=...` as the URL query arg, but the azure blob storage sdk returns the property as `blob.version_id`, so I think the existing adlfs PR which returns `info['version_id']` is correct for azure.

Standardizing it in the dvc-plugins is probably fine, and I pushed a change in the dvc-s3 PR that just sets both fields for S3 `info()`

Reply:
I think this should be standard across all filesystems regardless of their internal dependencies and implementations, although `info()` is left a lot upto the implementations.

Temporarily standardizing in dvc-plugins is also fine but it's too much effort. I'd be fine using [`_adapt_info`](https://github.com/iterative/dvc-data/blob/5654d2ba51bad6e951c20fe6c0e98266ea6920dc/src/dvc_data/hashfile/hash.py#L112) to standardize it.

Reply:
Didn't notice adapt_info, will just go ahead and use that for now